oauth2-useragent-core/src/test/java/com/microsoft/alm/oauth2/useragent/UserAgentImplTest.java

```diff
@@ -181,6 +183,28 @@ private static void assertLinesMatch(final String actual, final String... expect
         }
     }
 
+    @Test public void relayProperties_typical() throws Exception {
+        final HashSet<String> propertyNames = new HashSet<String>(Arrays.asList("http.proxyHost", "http.proxyPort"));
+        final Properties properties = new Properties();
+        properties.setProperty("http.proxyHost", "192.0.2.42");
+        final List<String> commands = new ArrayList<String>();
+
+        UserAgentImpl.relayProperties(properties, propertyNames, commands);
+
+        Assert.assertEquals(1, commands.size());
+        Assert.assertEquals("-Dhttp.proxyHost=192.0.2.42", commands.get(0));
+    }

```

Comment:
> craft an integration test that uses org.littleshoot:littleproxy:1.1.0-beta1 like the test I added in tfs-plugin that sets up a proxy server, connects to a remote host and then asserts that the proxy server was used

I'll be implementing this; please hold off on merging until I follow up.

Thanks!
\- Oli